### PR TITLE
Development of App's MVP

### PR DESCRIPTION
**Pull request by emmiiorji**
This request updates with the app's MVP

The following options are available.

- List all books
- List all music albums
- List all movies
- List of games
- List all genres 
- List all labels 
- List all authors 
- List all sources 
- Add a book
- Add a music album
- Add a movie
- Add a game

All the data is stored in JSON format.

Video demo for the app is linked [here](https://drive.google.com/file/d/1vUGpYE9g7JtkzQPH8iByrF38U8cGqQnu/view?usp=sharing).

### CODE REVIEW

File: item.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+require 'date'
+# date class for applying archive
+
+class Item
+  attr_accessor :title, :publish_date
+  attr_reader :id, :genre, :author, :label, :archived
```
Comment (Omar-Muhamad) [inline item.rb:6]: - [ ] Please add getters (attr_reader) for `@source`.

File: music_album.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  attr_accessor :title, :on_spotify
+
+  def initialize(*args, on_spotify)
+    super(*args)
+    @on_spotify = on_spotify
+  end
+
+  def can_be_archived?
+    super || @on_spotify
+  end
```
Comment (Omar-Muhamad) [inline music_album.rb:16]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things_divided.md#group-task:~:text=should%20return%20true%20if%20parent%27s%20method%20returns%20true%20AND%20if%20on_spotify%20equals%20true), Please make return true if parent's method returns true `AND` if on_spotify equals true.

File: game.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  def self.json_create(game)
+    new(game['id'], game['title'], game['publish_date'], game['archived'],
+        game['multiplayer'], game['last_played_at'],
+        game['author'], game['label'], game['genre'], game['source'])
+  end
+
+  def can_be_archived?()
+    super && last_played_at > 1
+  end
```
Comment (Omar-Muhamad) [inline game.rb:38]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things_divided.md#group-task:~:text=should%20return%20true%20if%20parent%27s%20method%20returns%20true%20AND%20if%20last_played_at%20is%20older%20than%202%20years), this method suppose to return true if parent's method returns true AND if last_played_at is older than 2 years. So please fix that.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi Team,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by rustam-tolipov — APPROVED** [review]
## Hello TEAM👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: item.rb (hunk 1 above)
Comment (waqaskanju) [reply item.rb]: image
consider it done.

**Pull request merged**